File: MvvmSample/app/src/main/java/com/abben/mvvmsample/CustomFragmentPagerAdapter.java
package com.abben.mvvmsample;

import java.util.ArrayList;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * Created by abben on 2017/5/3.
 */
public class CustomFragmentPagerAdapter extends FragmentPagerAdapter {
    private ArrayList<Fragment> mFragments;
    private String[] pageTitles = {"所有电影", "欧美电影", "日韩电影", "国产电影"};

    public CustomFragmentPagerAdapter(FragmentManager fm, ArrayList<Fragment> mFragments) {
        super(fm);
        this.mFragments = mFragments;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return pageTitles[position];
    }
}

File: MvvmSampleJetpack/app/src/main/java/com/infun/mvvmsamplejetpack/CustomFragmentPagerAdapter.kt
package com.infun.mvvmsamplejetpack

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 *  @author abben
 *  on 2019/1/17
 */
class CustomFragmentPagerAdapter(
    private val fm: FragmentManager, private val mFragments: List<Fragment>
) : FragmentPagerAdapter(fm) {

    private val pageTitles = arrayOf("所有电影", "欧美电影", "日韩电影", "国产电影")

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return pageTitles[position]
    }
}